frontend/src/searchBar/searchBar.component.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { Search } from '@mui/icons-material';
+import {
+  Avatar,
+  CircularProgress,
+  InputAdornment,
+  InputBase,
+  List,
+  ListItem,
+  ListItemText,
+  Popover,
+} from '@mui/material';
+import axios from 'axios';
+import debounce from 'lodash/debounce';
+
+import { Element } from '../interfaces/Element.component';
+import styles from './searchBar.module.css';
+
+interface SearchBarProps {
+  onSelect: (element: Element) => void;
+  isOpen: boolean;
+  setIsOpen: (open: boolean) => void;
+  searchTerm: string;
+  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const SearchBar: React.FC<SearchBarProps> = ({ onSelect, isOpen, setIsOpen, searchTerm, setSearchTerm }) => {
+  const [filteredResults, setFilteredResults] = useState<Element[]>([]);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isFocused, setIsFocused] = useState(false);
+
+  const fetchProducts = async (query: string) => {
+    if (!query.trim()) return [];
+    setIsLoading(true);
+
+    try {
+      const response = await axios.get('http://localhost:5000/product-search/search', {
+        params: { query, page: 1, limit: 50 },
+      });
+      return Array.isArray(response.data) ? response.data.slice(0, 50) : [];
+    } catch (error) {
+      console.error('Error fetching products:', error);
+      return [];
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const debouncedSearch = useCallback(
+    debounce(async (query: string) => {
+      setIsLoading(true);
+      setTimeout(async () => {
+        const results = await fetchProducts(query);
+        setFilteredResults(results);
+        setIsOpen(results.length > 0);
+        setIsLoading(false);
+      }, 1000);
+    }, 300),
+    [setIsOpen],
+  );
+
+  useEffect(() => {
+    debouncedSearch(searchTerm);
+  }, [searchTerm, debouncedSearch]);
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
There's no need of importing whole React class from library. Import only things you need.